cmd/main_test.go

```diff
@@ -267,3 +270,83 @@ var _ = AfterSuite(func() {
 		Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Error killing the server during test teardown: %v", err))
 	}
 })
+
+type mockAuthorization struct {
+	allowed bool
+	err error
+}
+
+func (m *mockAuthorization) runAccessCheck(
+	e *echo.Echo,
+	authCl authorizationv1Client.AuthorizationV1Interface,
+	email, namespace, resourceGroup, resource, verb string,
+) (bool, error) {
+	return m.allowed, m.err
+}
+
+var _ = Describe("TestGetNamespacesWithAccess", func() {
+	var (
+		e               *echo.Echo
+		c               echo.Context
+		authCl          *mockAuthorization
+		allNamespaces   []k8sapi.Namespace
+		expectedNs      []k8sapi.Namespace
+		actualNs        []k8sapi.Namespace
+		err             error
+	)
+
+	BeforeEach(func() {
+		e = echo.New()
+		req := httptest.NewRequest(http.MethodGet, "/", nil)
+		rec := httptest.NewRecorder()
+		c = e.NewContext(req, rec)
+		c.Request().Header.Set("test-email", "test@example.com")
+
+		authCl = &mockAuthorization{}
+		k8sClient, err = client.New(cfg, client.Options{Scheme: scheme})
+		allNamespaces = []{createNamespace(k8sClient, "test-ns1"), createNamespace(k8sClient, "test-ns1")}
+	})
+
+	JustBeforeEach(func() {
+		actualNs, err = getNamespacesWithAccess(e, c, authCl, allNamespaces)
+	})
+
+	Context("when access check allows all namespaces", func() {
+		BeforeEach(func() {
+			authCl.allowed = true
```

Comment:
Using envtest, we should create the namespaces\users\permissions instead of mocking the information.